server/routes/applications.js

```diff
@@ -96,15 +97,55 @@ router.get('/manage-applications/:appId/view/:env', (req, res, next) => {
   });
 });
 
-router.get('/manage-applications/:appId/update/:env', (req, res, next) => {
-  logger.info(`GET request to update application : ${req.path}`);
+router.get('/manage-applications/:appId/update/:env', (req, res) => {
+  logger.info(`GET request to serve update application page: ${req.path}`);
+  const id = req.params.appId;
+  const env = req.params.env;
   const viewData = {
-    this_data: null,
+    this_data: {
+      appId: id,
+      env: env
+    },
+    this_errors: null,
+    active_page: 'application-overview',
+    title: 'Edit application'
+  };
+
+  applicationsDeveloperService.getApplication(id, env)
+    .then(appData => {
+      viewData.this_data.applicationName = appData.data.name;
+      viewData.this_data.description = appData.data.description;
+      viewData.this_data.privacyPolicy = appData.data.privacy_policy_url;
+      viewData.this_data.terms = appData.data.terms_and_conditions_url;
+      console.log('APP DATA: ', appData);
+      res.render(`${routeViews}/edit.njk`, viewData);
+    }).catch(err => {
+      viewData.this_errors = routeUtils.processException(err);
+      res.render(`${routeViews}/edit.njk`, viewData);
+    });
+});
+
+router.post('/manage-applications/:appId/update/:env', (req, res) => {
+  logger.info(`PUT request to update the application: ${req.path}`);
+  const payload = req.body;
+  payload.env = req.params.env;
+  payload.appId = req.params.appId;
+  const viewData = {
+    this_data: payload,
     this_errors: null,
     active_page: 'application-overview',
-    title: 'Manage Application'
+    title: 'Update an application'
   };
-  res.render(`${routeViews}/edit.njk`, viewData);
+  validator.validateApplication(payload, true)
+    .then(_ => {
+      return applicationsDeveloperService.updateApplication(payload);
```

Comment:
I'd rather see things passed in as specific function parameters. I don't like that we have to assume the structure of the 'payload' in the update method and we don't have anything to ensure that values are passed in as expected.

Reply:
So updateApplication( payload, appId, env)
This is more inline with how its been done in other methods of the service class.

Reply:
Agreed to disagree. And not refactor other functions to match